src/k8s/kubernetes.py

```
@@ -36,29 +56,49 @@ def create_harbor_secret(host_address: str,
     try:
         core_client.create_namespaced_secret(namespace=namespace, body=secret)
     except client.exceptions.ApiException:
+        logger.warning(f"Harbor secret already exists in namespace {namespace}, attempting to resolve conflict by "
+                       f"deleting and recreating the secret.")
         try:
             core_client.delete_namespaced_secret(name=name, namespace=namespace)
             core_client.create_namespaced_secret(namespace=namespace, body=secret)
         except client.exceptions.ApiException as e:
             if e.reason != 'Conflict':
-                raise e
+                logger.error(f"Unknown error during harbor secret creation: {repr(e)}")
```

Ruff (0.15.10)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)